app/models/ext_management_system.rb

```diff
@@ -686,7 +686,7 @@ def wait_for_ems_workers_removal
   end
 
   def orchestrate_destroy_queue(task_id, queue_options = {})
-    self.class._queue_task('orchestrate_destroy', [id], task_id, queue_options)
+    self.class._queue_task('orchestrate_destroy', [id], task_id, queue_options.reverse_merge(:msg_timeout => 3_600, :priority => MiqQueue::MEDIUM_PRIORITY))
```

Comment:
My only question is if we actually want EMS destroy to run at a higher priority than most other operations?  Once it is paused is there a greater need for it to complete sooner than say a credential validation on a provider creation or a vm power operation?

Reply:
Considering destroys are likely coming from users, I think it makes sense to give them higher than normal priority.  Maybe not high (priority worker) level but higher than normal.

Reply:
Right both of those others are user initiated as well.

Reply:
I agree with @agrare that I think normal priority is sufficient.  The pause should stay high, but the rest can happen when it happens.

Perhaps we want to set expectations better in the UI message after you press delete.  Right now it just says something like initiating delete, but perhaps we can add that the provider will be paused and the delete will happen over some period of time.

Reply:
Perhaps a vm operation can affect other features of MIQ but I think it's clearly different and less impactful than a complete new provider cred check and deleting a provider.  But, even the vm operation, if initiated by a user, perhaps should also be in a user initiated priority that is also slightly higher than normal priority.

Regardless though...credential verification should be higher than normal priority because, just like destroy, there are so many other features that depend on it.  
We already do the auth check on the schedule worker at high priority for this same reason.

```
  def host_authentication_check_schedule
    queue_work_on_each_zone(:class_name  => "Host", :method_name => "authentication_check_schedule", :priority => MiqQueue::HIGH_PRIORITY)
  end

  def ems_authentication_check_schedule
    queue_work_on_each_zone(:class_name  => "ExtManagementSystem", :method_name => "authentication_check_schedule", :priority => MiqQueue::HIGH_PRIORITY)
  end
```

Now, I'm not saying we need to do high priority here and use up a priority worker.  Only that work that is depended on by several key features should be run in a slightly higher priority so it's not sitting behind a raise "worker started" event.

Destroy of a provider is basically removing an auth + all related entities it knows about and their metrics/data.  The problem with delaying destroy is the amount of work that continues to be processed for an entity that has not yet been removed because it's sitting in line behind other normal priority work.  This may include the inventory work that actually resurrects objects that we just tried to delete.  

Destroys shouldn't be slow and it's a bug if they are.

Reply:
Ok, I'll just fix the bug with msg_timeout not being honored in the "requeue if not yet paused" case.